src/main/java/uk/gov/companieshouse/api/strikeoffobjections/controller/ObjectionRequestController.java

```diff
@@ -1,77 +1,75 @@
 import uk.gov.companieshouse.service.rest.response.PluggableResponseEntityFactory;
 
-import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.Map;
 
 @RestController
-@RequestMapping(value = "/company/{companyNumber}/objections/strike-off")
+@RequestMapping(value = "/company/{companyNumber}/strike-off-objections")
 public class ObjectionRequestController {
 
     private static final String LOG_COMPANY_NUMBER_KEY = LogConstants.COMPANY_NUMBER.getValue();
-    private static final String LOG_REQUEST_ID_KEY = LogConstants.REQUEST_ID.getValue();
-    private static final String LOG_REASON_KEY = LogConstants.REASON.getValue();
     private static final String ERIC_REQUEST_ID_HEADER = "X-Request-Id";
 
     private PluggableResponseEntityFactory responseEntityFactory;
+    private IObjectionService objectionService;
     private ApiLogger apiLogger;
 
     @Autowired
-    public ObjectionRequestController(
-            PluggableResponseEntityFactory responseEntityFactory,
-            ApiLogger apiLogger) {
-
+    public ObjectionRequestController(PluggableResponseEntityFactory responseEntityFactory, IObjectionService objectionService, ApiLogger apiLogger) {
         this.responseEntityFactory = responseEntityFactory;
+        this.objectionService = objectionService;
         this.apiLogger = apiLogger;
     }
 
-    @PostMapping("/request")
-    public ResponseEntity<ChResponseBody<ObjectionResponseBody>> createObjectionRequest(
-            @Valid @RequestBody ObjectionRequestBody requestBody,
+    @PostMapping
+    public ResponseEntity<ChResponseBody<ObjectionResponse>> createObjection(
             @PathVariable("companyNumber") String companyNumber,
-            @RequestHeader(value = ERIC_REQUEST_ID_HEADER) String requestId) {
-        
+            @RequestHeader(value = ERIC_REQUEST_ID_HEADER) String requestId
+    ) {
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(LOG_COMPANY_NUMBER_KEY, companyNumber);
+
         apiLogger.infoContext(
-            requestId,
-            String.format("POST /request received: %s %s, %s %s, %s %s",
-                LOG_REQUEST_ID_KEY, requestId, LOG_COMPANY_NUMBER_KEY, companyNumber, LOG_REASON_KEY, requestBody.getReason()));
+                requestId,
+                "POST / request received",
+                logMap
+        );
 
         try {
-            // TODO Implement actual business logic here when stories are available
-            
-            ObjectionResponseBody objectionResponseBody = new ObjectionResponseBody();
-            objectionResponseBody.setRequestId("12345");
-            
-            return responseEntityFactory.createResponse(ServiceResult.created(objectionResponseBody));
-
-        } catch (Exception e) {            
+            String objectionId = objectionService.createObjection(requestId, companyNumber);
+            ObjectionResponse response = new ObjectionResponse(objectionId);
```

Comment:
do we need to create an ObjectionResponse object or can we just return the string?

Reply:
Just returning the string wouldn't conform to the spec which wants `"id" : "OBJECTION_ID"`
I could possibly return a map to achieve the same effect

Reply:
ok I forgot about the 2 parts